meteor/server/api/ingest/rundownInput.ts

```diff
@@ -417,24 +428,83 @@ function updateRundownFromIngestData (
 	})
 
 	// Prepare updates:
-	const prepareSaveSegments = prepareSaveIntoDb(Segments, {
+	let prepareSaveSegments = prepareSaveIntoDb(Segments, {
 		rundownId: rundownId
 	}, segments)
-	const prepareSaveParts = prepareSaveIntoDb<Part, DBPart>(Parts, {
+	let prepareSaveParts = prepareSaveIntoDb<Part, DBPart>(Parts, {
 		rundownId: rundownId,
 	}, parts)
-	const prepareSavePieces = prepareSaveIntoDb<Piece, Piece>(Pieces, {
+	let prepareSavePieces = prepareSaveIntoDb<Piece, Piece>(Pieces, {
 		rundownId: rundownId,
 		dynamicallyInserted: { $ne: true } // do not affect dynamically inserted pieces (such as adLib pieces)
 	}, segmentPieces)
-	const prepareSaveAdLibPieces = prepareSaveIntoDb<AdLibPiece, AdLibPiece>(AdLibPieces, {
+	let prepareSaveAdLibPieces = prepareSaveIntoDb<AdLibPiece, AdLibPiece>(AdLibPieces, {
 		rundownId: rundownId,
 	}, adlibPieces)
 
-	// determine if update is allowed here
-	if (!isUpdateAllowed(dbPlaylist, dbRundown, { changed: [{ doc: dbRundown, oldId: dbRundown._id }] }, prepareSaveSegments, prepareSaveParts)) {
-		ServerRundownAPI.unsyncRundown(dbRundown._id)
-		return false
+	if (Settings.allowUnsyncedSegments) {
+
+		if (!isUpdateAllowed(dbPlaylist, dbRundown, { changed: [{ doc: dbRundown, oldId: dbRundown._id }] })) {
+			ServerRundownAPI.unsyncRundown(dbRundown._id)
+			return false
+		} else {
+			const segmentChanges: SegmentChanges[] = splitIntoSegments(
```

Comment:
This feels to have to split the data back into segments. I suppose it has to be done though, unless we do the prepare step again.

Has this been tested for any cross-bleed issues?
As in, moving piece99 from segment1 to segment2. I assume this should be safe due to how we do ids, but that would be good to confirm.

Reply:
I have not managed to test that use case, as the iNews gateway does not have that awareness of parts / pieces being moved / copied (it's not how iNews works as such). I can say that it handles parts within a segment changing order with the iNews workflow. I suspect an ENPS system will be needed to test every permutation.